packages/fxa-payments-server/src/routes/Product/SubscriptionCreate/index.tsx

```diff
@@ -502,7 +527,17 @@ async function handlePaymentIntent({
   }
 }
 
-const hasExistingCard = (customer: Customer | null) =>
-  customer && customer.last4 && customer.subscriptions.length > 0;
+const isExistingStripeCustomer = (customer: Customer | null) =>
+  customer &&
+  customer.payment_provider === 'stripe' &&
+  customer.subscriptions.length > 0;
+
+const isExistingPaypalCustomer = (customer: Customer | null) =>
+  customer &&
+  customer.payment_provider === 'paypal' &&
+  customer.subscriptions.length > 0;
+
+const isNewCustomer = (customer: Customer | null) =>
+  !isExistingStripeCustomer(customer) && !isExistingPaypalCustomer(customer);
```

Comment:
The way this is written requires future devs to know to add to it when they add a third payment provider.  It'd be easier on them if it checks for the lack of any payment provider and not two specific payment providers.